nimble/host/include/host/ble_gap.h

```diff
@@ -913,6 +966,28 @@ int ble_gap_ext_adv_stop(uint8_t instance);
 int ble_gap_ext_adv_set_data(uint8_t instance, struct os_mbuf *data);
 int ble_gap_ext_adv_rsp_set_data(uint8_t instance, struct os_mbuf *data);
 int ble_gap_ext_adv_remove(uint8_t instance);
+
+#if MYNEWT_VAL(BLE_PERIODIC_ADV)
+/* Periodic Advertising */
+int ble_gap_per_adv_configure(uint8_t instance, const struct ble_gap_per_adv_params *params);
+int ble_gap_per_adv_start(uint8_t instance);
+int ble_gap_per_adv_stop(uint8_t instance);
+int ble_gap_per_adv_set_data(uint8_t instance, struct os_mbuf *data);
+int ble_gap_per_adv_create_sync (uint8_t filter_policy,
```

Comment:
Thanks for the note.

The idea is that sync established event can never be received without having extended scanning first !

The point here is that even if the application is only interested in establishing a sync with a periodic advertising, this can never happen without having an extended scanning enabled at the first place, so ble_gap_ext_disc will have been called before, so the gap event callback should be already set.

After the sync is established, then the application can disable extended scanning and live only with periodic scanning. At this time, the callback is already set since the time of extended scanning.

Reply:
Sync established event provides no data at all. It only confirms that the sync with a certain advertising SID is established, and provides some info about this sync, like advertising PHY and advertising interval.

I believe what you are looking for is providing a way for the app to receive periodic advertising data. If this is the case, there is already the periodic advertising reports that the scanner app receives periodically from the advertiser.

Please let me know if I missed your point !